src/resolver.h

```diff
@@ -22,40 +15,35 @@ class UsdOpenAssetIOResolver final : public PXR_NS::ArDefaultResolver {
  protected:
   /* Ar Resolver Implementation */
   [[nodiscard]] std::string _CreateIdentifier(
-      const std::string &assetPath, const PXR_NS::ArResolvedPath &anchorAssetPath) const final;
+      const std::string &assetPath, const PXR_NS::ArResolvedPath &anchorAssetPath) const override;
 
   [[nodiscard]] std::string _CreateIdentifierForNewAsset(
-      const std::string &assetPath, const PXR_NS::ArResolvedPath &anchorAssetPath) const final;
+      const std::string &assetPath, const PXR_NS::ArResolvedPath &anchorAssetPath) const override;
 
-  [[nodiscard]] PXR_NS::ArResolvedPath _Resolve(const std::string &assetPath) const final;
+  [[nodiscard]] PXR_NS::ArResolvedPath _Resolve(const std::string &assetPath) const override;
 
   [[nodiscard]] PXR_NS::ArResolvedPath _ResolveForNewAsset(
-      const std::string &assetPath) const final;
+      const std::string &assetPath) const override;
 
   /* Asset Operations*/
-  [[nodiscard]] std::string _GetExtension(const std::string &assetPath) const final;
+  [[nodiscard]] std::string _GetExtension(const std::string &assetPath) const override;
 
   [[nodiscard]] PXR_NS::ArAssetInfo _GetAssetInfo(
-      const std::string &assetPath, const PXR_NS::ArResolvedPath &resolvedPath) const final;
+      const std::string &assetPath, const PXR_NS::ArResolvedPath &resolvedPath) const override;
 
   [[nodiscard]] PXR_NS::ArTimestamp _GetModificationTimestamp(
-      const std::string &assetPath, const PXR_NS::ArResolvedPath &resolvedPath) const final;
+      const std::string &assetPath, const PXR_NS::ArResolvedPath &resolvedPath) const override;
 
   [[nodiscard]] std::shared_ptr<PXR_NS::ArAsset> _OpenAsset(
-      const PXR_NS::ArResolvedPath &resolvedPath) const final;
+      const PXR_NS::ArResolvedPath &resolvedPath) const override;
 
   [[nodiscard]] bool _CanWriteAssetToPath(const PXR_NS::ArResolvedPath &resolvedPath,
-                                          std::string *whyNot) const final;
+                                          std::string *whyNot) const override;
 
   [[nodiscard]] std::shared_ptr<PXR_NS::ArWritableAsset> _OpenAssetForWrite(
-      const PXR_NS::ArResolvedPath &resolvedPath, WriteMode writeMode) const final;
+      const PXR_NS::ArResolvedPath &resolvedPath, WriteMode writeMode) const override;
 
  private:
-  /// OpenAssetIO Logger. Uses TF Log/Error functions.
-  openassetio::log::LoggerInterfacePtr logger_;
-  /// OpenAssetIO Manager. Initialised on construction.
-  openassetio::hostApi::ManagerPtr manager_;
-
-  /// Fixed OpenAssetIO context.
-  openassetio::ContextConstPtr context_;
+  class Impl;
+  std::unique_ptr<Impl> impl_;
```

Comment:
[Consider] You spoke to me about how this was not neccesary. If there is no need for pimpl, why do it? 

It's your code at the end of the day, but all this accomplishes to my mind is :
- Create a much larger changeset than necessary
- Force a redundant heap allocation
- Mislead people into thinking there is a need to pimpl for interface stability reasons
- Make the code harder to understand for people who haven't encountered pimpl before

If your goal was to be "clean", ie, separating the deferment of failed results to the default resolver from the main implementation. That could be accomplished in a more direct way with free functions.

Reply:
I did roll back the pimpl, but then realised I'd end up `#include`ing  `FileUrlPathConverter` to add it as a new member, whereas other types were forward-declared in the header. A minor issue for sure, but it made me reflect on the style and I ended up liking the pimpl more, since it keeps the header completely clean, with no reference to OpenAssetIO at all.

I wonder now though if there is any reason to have a header at all, perhaps that's the real misleading thing...

Reply:
OK, completely refactored and force-pushed so there is no confusing interface header. Also split things up a bit better commit-wise